src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/net/NetServer.java

```diff
@@ -138,12 +138,13 @@ private Server buildHttpsServer() {
   @Override
   public void subscribe(
       final SubscribeMessage request, final StreamObserver<SubscribeResponse> responseObserver) {
-    LOG.debug("subscribe received");
     if (subscribeHandler != null) {
       subscribeHandler.handleSubscriptionRequest(request, responseObserver);
+    } else {
+      LOG.error("Subscribe request received before handler is set.");
```

Comment:
This will only happen during initial switch on phase of RCA where the framework comes up on master first before being turned on in the data nodes. After that we should not be seeing this. However, I can add buffering as an improvement - will create an issue around it.

Reply:
cool.